### DISCUSSION TITLE
Project on ISMAGS library incorporation

### DISCUSSION THREAD

**Discussion by BasLaa** [networkx/networkx#6447] [General]
Hi,
I saw this project on the ideas page of the website and was interested in it, I believe the suggested mentor was @dschult . I started contributing to NetworkX last year and looked into a different project on graph isomorphism, but due to circumstances ended up not being able to go further into it. I believe that this year NetworkX falls under NumFocus again for GSoC?

For this project I started looking into the new Python code by sandia labs for ISMAGS, and started looking into the original paper. One thing I was wondering is what the project itself would actually entail. It says the outcome is a plan for how to incorporate this new library into NetworkX, does that also include actually executing the plan and somehow combining the code, or is the project result only the planning itself? I'm curious to hear more!
Bas

**Comment by BasLaa** [comment]
Is there someone who could give an update on this?

**Comment by dschult** [comment]
It would be great if a plan could be developed and implemented. And probably any planning effort will include some attempts at implementing it to see what the issues are and/or where difficulties would arise. So, a goal should probably be to implement this transition at least to some extent. But that might be a bigger project than can be completed in a summer. I'm not sure because I haven't looked at it too hard yet. The final project outcome is an implemented transfer of the code and a more uniform approach to isomorphism API in NetworkX in general.
